pkg/gpu/nvidia/health_check/health_checker.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthcheck
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/container-engine-accelerators/pkg/gpu/nvidia/util"
+	"github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml"
+	"github.com/golang/glog"
+
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+// GPUHealthChecker checks the health of nvidia GPUs.
+type GPUHealthChecker struct {
+	devices     map[string]pluginapi.Device
+	nvmlDevices map[string]*nvml.Device
+	health      chan pluginapi.Device
+	eventSet    nvml.EventSet
+	stop        chan bool
+}
+
+// NewGPUHealthChecker returns a GPUHealthChecker object for a given device name
+func NewGPUHealthChecker(devices map[string]pluginapi.Device, health chan pluginapi.Device) *GPUHealthChecker {
+	return &GPUHealthChecker{
+		devices:     devices,
+		nvmlDevices: make(map[string]*nvml.Device),
+		health:      health,
+		stop:        make(chan bool),
+	}
+}
+
+// Start registers NVML events and starts listening to them
+func (hc *GPUHealthChecker) Start() error {
+	glog.Info("Starting GPU Health Checker")
+
+	// Building mapping between device ID and their nvml represetation
+	count, err := nvml.GetDeviceCount()
+	if err != nil {
+		return fmt.Errorf("failed to get device count: %s", err)
+	}
+
+	glog.Infof("Found %d GPU devices", count)
+	for i := uint(0); i < count; i++ {
+		device, err := nvml.NewDeviceLite(i)
+		if err != nil {
+			return fmt.Errorf("failed to read device with index %d: %v", i, err)
+		}
+		deviceName, err := util.DeviceNameFromPath(device.Path)
+		if err != nil {
+			glog.Errorf("Invalid GPU device path found: %s. Skipping this device", device.Path)
+		}
+
+		if _, ok := hc.devices[deviceName]; !ok {
+			// we only monitor the devices passed in
+			glog.Warningf("Ignoring device %s for health check.", deviceName)
+			continue
+		}
+
+		glog.Infof("Found device %s for health monitoring. UUID: %s", deviceName, device.UUID)
+		hc.nvmlDevices[deviceName] = device
```

Comment:
So my understanding is that when MIG mode is enabled, `hc.nvmlDevices` will be empty, because device name we discover from nvml won't match the MIG device names we use (e.g. `gpu0/gi1`) on line 68. 

So this means when MIG mode is enabled and GPUs are partitioned, there will be no health checks?  That's fine for now, but just want to confirm the behavior.

Reply:
your understanding is correct. If `DeviceNameFromPath(migPath)` does not match the device names passed in, then the mig device will be ignored.